client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -33,10 +33,22 @@ module.exports = React.createClass( {
 		}
 
 		window.scrollTo( 0, 0 );
-		analytics.tracks.recordEvent( 'calypso_checkout_page_view', { saved_cards: this.props.cards.get().length } );
+		if ( this.props.cart.hasLoadedFromServer ) {
+			this.trackPageView( this.props.cart );
```

Comment:
This won't work, since when we call it from `componentWillUpdate` this.props will still be the old props.